sdk/typescript/telemetry/telemetry.ts

```diff
@@ -0,0 +1,222 @@
+    this.sdk = sdk
+  }
+
+  /**
+   * Shutdown the Open Telemetry SDK to flush traces and metrics and close the connection.
+   */
+  async close() {
+    if (this.sdk) {
+      this.sdk.shutdown()
+    }
+  }
+
+  configure() {
+    if (!otelConfigured()) {
+      console.debug("Telemetry not configured")
+      return
+    }
+
+    if (!otelEnabled()) {
+      console.debug("Telemetry disabled")
+      return
+    }
+
+    this.setupEnv()
+  }
+
+  /**
+   * Setup environment for auto-configuring the SDK.
+   */
+  setupEnv() {
+    // Setup a default value for the service name.
+    if (!process.env[OtelEnv.OTEL_SERVICE_NAME]) {
+      process.env[OtelEnv.OTEL_SERVICE_NAME] = SERVICE_NAME
+    }
+
+    // TODO: The insecure flag should be set by the shim instead of the SDK.
+    const endpoints = [
+      OtelEnv.OTEL_EXPORTER_OTLP_ENDPOINT,
+      OtelEnv.OTEL_EXPORTER_OTLP_METRICS_ENDPOINT,
+      OtelEnv.OTEL_EXPORTER_OTLP_LOGS_ENDPOINT,
+      OtelEnv.OTEL_EXPORTER_OTLP_TRACES_ENDPOINT,
+    ]
+
+    endpoints.forEach((endpoint) => {
+      const env = process.env[endpoint]
+      if (env && env.startsWith("http://")) {
+        process.env[OtelEnv.OTEL_EXPORTER_OTLP_INSECURE] = "true"
+      }
+    })
+  }
+}
+
+const configurator = new DaggerOtelConfigurator()
+
+export function initiliaze() {
+  configurator.initialize()
+}
+
+export async function close() {
+  await configurator.close()
+}
+
+/**
+ * Return a tracer to use with Dagger.
+ *
+ * The tracer is automatically initialized if not already done.
+ * As a conveniance function, you can use `withTracingSpan` that automatically close
+ * the span at the end of the function.
+ */
+export function getTracer(): opentelemetry.Tracer {
+  initiliaze()
+
+  const tracer = opentelemetry.trace
+    .getTracerProvider()
+    .getTracer("dagger-typescript-sdk")
+
+  return tracer
+}
+
+export function initContext() {
+  const ctx = opentelemetry.context.active()
+
+  const parentID = process.env.TRACEPARENT
+  if (parentID) {
+    return opentelemetry.propagation.extract(ctx, {
+      traceparent: parentID,
+    })
+  }
+
+  return ctx
+}
+
+/**
+ * Execute the functions with a custom span with the given name using startActiveSpan.
+ * The function executed will use the parent context of the function (it can be another span
+ * or the main function).
+ *
+ * @param name The name of the span
+ * @param fn The functions to execute
+ *
+ * WithTracingSpan returns the result of the executed functions.
+ *
+ * The span is automatically ended when the function is done.
+ * The span is automatically marked as an error if the function throws an error.
+ *
+ *
+ * @example
+ * ```
+ * return withTracingSpan(name, async () => {
+ *   return this.containerEcho("test").stdout()
+ * })
+ * ```
+ */
+export async function withTracingSpan<T>(
+  name: string,
+  fn: (span: opentelemetry.Span) => Promise<T>,
+): Promise<T> {
+  return await opentelemetry.context.with(
+    opentelemetry.context.active(),
```

Comment:
This is not necessary as it defaults to the active context already:

https://github.com/open-telemetry/opentelemetry-js/blob/3ab4f765d8d696327b7d139ae6a45e7bd7edd924/packages/opentelemetry-sdk-trace-base/src/Tracer.ts#L237

Reply:
Fixed